packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -1028,11 +1092,18 @@ class _RenderSegmentedControl<T extends Object> extends RenderBox
   void paint(PaintingContext context, Offset offset) {
     final List<RenderBox> children = getChildrenAsList();
 
+    // Children contains both segment and separator and the order is segment ->
+    // separator -> segment. So to paint separators, indes should start from 1 and
+    // the step should be 2.
     for (int index = 1; index < childCount; index += 2) {
       _paintSeparator(context, offset, children[index]);
     }
 
     final int? highlightedChildIndex = highlightedIndex;
+    final bool isLeadingChild = highlightedChildIndex != null
```

Comment:
uber nit: the null check is not needed?

Reply:
It would be possible that groupValue is null, and we don't have highlighted child index, right?

Reply:
if `highlightedChildIndex == 0` then it can't be null.

Reply:
That's right! Thanks for catching this! Updated.